src/fs/metadata_log.cc

```diff
@@ -152,37 +192,53 @@ void metadata_log::memory_only_delete_inode(inode_t inode) {
         [](const inode_info::directory& dir) {
             assert(dir.entries.empty());
         },
-        [](const inode_info::file&) {
-            // TODO: for compaction: update used inode_data_vec
+        [&](inode_info::file& file) {
+            cut_out_data_range(file, {
+                .beg = 0,
+                .end = std::numeric_limits<decltype(file_range::end)>::max()
+            });
+            _compacted_log_size -= ondisk_entry_size<ondisk_truncate>(); // See memory_only_create_inode() for why it is here
         }
     }, it->second.contents);
 
     _inodes.erase(it);
+    _compacted_log_size -= ondisk_entry_size<ondisk_create_inode>();
 }
 
 void metadata_log::memory_only_small_write(inode_t inode, file_offset_t file_offset, temporary_buffer<uint8_t> data) {
     inode_data_vec data_vec = {
         {file_offset, file_offset + data.size()},
         inode_data_vec::in_mem_data {std::move(data)}
     };
-
     auto it = _inodes.find(inode);
     assert(it != _inodes.end());
     assert(it->second.is_file());
     write_update(it->second.get_file(), std::move(data_vec));
+    _compacted_log_size += ondisk_entry_size<ondisk_small_write_header>(data.size());
 }
 
 void metadata_log::memory_only_disk_write(inode_t inode, file_offset_t file_offset, disk_offset_t disk_offset,
         size_t write_len) {
+    assert(write_len <= _cluster_size);
     inode_data_vec data_vec = {
         {file_offset, file_offset + write_len},
         inode_data_vec::on_disk_data {disk_offset}
     };
-
     auto it = _inodes.find(inode);
     assert(it != _inodes.end());
     assert(it->second.is_file());
     write_update(it->second.get_file(), std::move(data_vec));
+    _compacted_log_size += (write_len == _cluster_size ?
+        ondisk_entry_size<ondisk_large_write>() :
+        ondisk_entry_size<ondisk_medium_write>());
+    cluster_id_t cluster_id = offset_to_cluster_id(disk_offset, _cluster_size);
+    assert(_read_only_data_clusters.find(cluster_id) == _read_only_data_clusters.end());
+    auto [cc_info_it, created] = _writable_data_clusters.try_emplace(cluster_id);
+    auto& cc_info = cc_info_it->second;
+    cc_info.add_data(disk_offset, inode, {file_offset, file_offset + write_len});
+    if (created) {
+        _data_cluster_contents_info_map[cluster_id] = &cc_info;
```

Comment:
We could put an assert here that this is creating an element indeed.
```suggestion
        auto& [_, inserted] = _data_cluster_contents_info_map.try_emplace(cluster_id, &cc_info);
        assert(inserted);
```